lib/wasi/src/bin_factory/mod.rs

```diff
@@ -0,0 +1,105 @@
+use std::{
+    collections::HashMap,
+    ops::Deref,
+    sync::{Arc, RwLock},
+};
+
+use wasmer_vfs::AsyncReadExt;
+
+mod binary_package;
+mod exec;
+mod module_cache;
+
+pub(crate) use exec::SpawnedProcess;
+use sha2::*;
+
+pub use self::{
+    binary_package::*,
+    exec::{spawn_exec, spawn_exec_module},
+    module_cache::ModuleCache,
+};
+use crate::{os::command::Commands, WasiRuntimeImplementation, WasiState};
+
+#[derive(Debug, Clone)]
+pub struct BinFactory {
```

Comment:
What is `BinFactory` really?

Reply:
I was about to ask the same question. Is that related to the spawn syscall, to handle new binary being spawned or something like that?

Reply:
In particular it handles bundling `webc` packages, including merging dependencies.

Reply:
I have a partial rewrite of this to make it cleaner, but that should be merged after the main PR.